text/0074-stateful-multisig-pallet.md

```diff
@@ -0,0 +1,585 @@
+
+Quick review over the extrinsics for both as it affects the block size:
+
+Stateless Multisig:
+Both `as_multi` and `approve_as_multi` has a similar parameters:
+
+```rust
+origin: OriginFor<T>,
+threshold: u16,
+other_signatories: Vec<T::AccountId>,
+maybe_timepoint: Option<Timepoint<BlockNumberFor<T>>>,
+call_hash: [u8; 32],
+max_weight: Weight,
+```
+
+Stateful Multisig:
+We have the following extrinsics:
+
+```rust
+pub fn start_proposal(
+			origin: OriginFor<T>,
+			multisig_account: T::AccountId,
+			call_hash: T::Hash,
+		)
+```
+
+```rust
+pub fn approve(
+			origin: OriginFor<T>,
+			multisig_account: T::AccountId,
+			call_hash: T::Hash,
+		)
+```
+
+```rust
+pub fn execute_proposal(
+			origin: OriginFor<T>,
+			multisig_account: T::AccountId,
+			call: Box<<T as Config>::RuntimeCall>,
+		)
+```
+
+The main takeway is that we don't need to pass the threshold and other signatories in the extrinsics. This is because we already have the threshold and signatories in the state (only once).
+
+So now for the caclulations, given the following:
+
+* K is the number of multisig accounts.
+* N is number of signers in each multisig account.
+* For each proposal we need to have 2N/3 approvals.
+
+The table calculates if each of the K multisig accounts has one proposal and it gets approved by the 2N/3 and then executed. How much did the total Blocks and States sizes increased by the end of the day.
+
+Note: We're not calculating the cost of proposal as both in statefull and stateless multisig they're almost the same and gets cleaned up from the state once the proposal is executed or canceled.
+
+Stateless effect on blocksizes = 2/3*K*N^2 (as each user of the 2/3 users will need to call approve_as_multi with all the other signatories(N) in extrinsic body)
+
+Stateful effect on blocksizes = K * N (as each user will need to call approve with the multisig account only in extrinsic body)
+
+Stateless effect on statesizes = Nil (as the multisig account is not stored in the state)
+
+Stateful effect on statesizes = K*N (as each multisig account (K) will be stored with all the signers (K) in the state)
+
+| Pallet         |  Block Size   | State Size |
+|----------------|:-------------:|-----------:|
+| Stateless      |     2/3*K*N^2 |        Nil |
+| Stateful       |          K*N  |       K*N  |
+
+Simplified table removing K from the equation:
+| Pallet         |  Block Size   | State Size |
+|----------------|:-------------:|-----------:|
+| Stateless      |           N^2 |        Nil |
+| Stateful       |           N   |         N  |
+
+So even though the stateful multisig has a larger state size, it's still more efficient in terms of block size and total footprint on the blockchain.
+
+### Ergonomics
+
+The Stateful Multisig will have better ergonomics for managing multisig accounts for both developers and end-users.
```

Comment:
Would be great addition to have a better functional comparison to the stateless pallet instead of a generic statement. Are there use cases where you believe stateless is better?

Reply:
> Are there use cases where you believe stateless is better?

That's a good question. I don't have any use case in my mind that the stateless is better TBH. Stateful can do all what the stateless do with added functionality, fine tuned control and clearer API.  Do you have a case in mind?

> Would be great addition to have a better functional comparison to the stateless pallet instead of a generic statement.

I thought the RFC itself is an enough comparison to make it clear by reaching this point. Do you suggest explaining it again like in overview and motivation sections or add something else?